es/es-server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -134,6 +134,15 @@
         Setting.byteSizeSetting("index.translog.retention.size", new ByteSizeValue(512, ByteSizeUnit.MB), Property.Dynamic,
             Property.IndexScope);
 
+    /**
+     * The minimum size of a merge that triggers a flush in order to free resources
+     */
+    public static final Setting<ByteSizeValue> INDEX_FLUSH_AFTER_MERGE_THRESHOLD_SIZE_SETTING =
```

Comment:
shall we also add and document it as a table setting?

Reply:
As long a it is not part of the supported table parameters a user wouldn't be able to change it. I would not expose it right now, assuming the default as a good fit.

Reply:
do you thing we need a changelog for this change?